jax/_src/config.py

```diff
@@ -376,7 +376,7 @@ def define_bool_state(
   an error.
   """
   if not isinstance(default, bool):
-    raise TypeError(f"Default value must be of type bool, got {default}")
+    raise TypeError(f"Default value must be of type bool, got {default} of type {type(default)}")
```

Comment:
I think a better thing here would be `getattr(type(default), '__name__', type(default))` so that we always report something even if __name__ doens't exist on the type.